src/ue4docker/test.py

```diff
@@ -14,45 +24,52 @@ def test():
     if len(sys.argv) > 1 and sys.argv[1].strip("-") not in ["h", "help"]:
         # Verify that the specified container image exists
         tag = sys.argv[1]
-        image = GlobalConfiguration.resolveTag(
+        image_name = GlobalConfiguration.resolveTag(
             "ue4-full:{}".format(tag) if ":" not in tag else tag
         )
-        if DockerUtils.exists(image) == False:
+
+        try:
+            image = client.images.get(image_name)
+        except ImageNotFound:
             logger.error(
                 'Error: the specified container image "{}" does not exist.'.format(
-                    image
+                    image_name
                 )
             )
             sys.exit(1)
 
         # Use process isolation mode when testing Windows containers, since running Hyper-V containers don't currently support manipulating the filesystem
-        platform = ImageUtils.image_platform(client, image)
+        platform = image.attrs["Os"]
         isolation = "process" if platform == "windows" else None
 
         # Start a container to run our tests in, automatically stopping and removing the container when we finish
         logger.action(
-            'Starting a container using the "{}" image...'.format(image), False
+            'Starting a container using the "{}" image...'.format(image_name), False
+        )
+        container = ContainerUtils.start_for_exec(
+            client, image_name, platform, isolation=isolation
         )
-        container = ContainerUtils.start_for_exec(client, image, isolation=isolation)
         with ContainerUtils.automatically_stop(container):
```

Comment:
Probably for a future change, but idiomatically this should be one call, something like `with container = ContainuerUtils.exec_container(client, image_name, platform, isolation=isolation):`. One of the strengths of the context manager pattern is that it provides a way to guarantee a resource-cleanup opportunity if something inside `start_for_exec` throws an exception.

I also could be misremembering something we built locally in my last job, but I thought the docker client already had a context manager pattern for this kind of use-case.

Reply:
Yep, I'm not happy with ergonomics of current API, but since we have only a single (and very rarely used) place where it appears, possibly it can live as it is for some time.